### ISSUE TITLE
Dictionary Comprehension Crash

### ISSUE THREAD

**Issue by wrywerytwreywery** [scopatz/xonsh#210]
xonsh just crashed when running the following:
```
$ x = {k:float(v) for k,v in x.iteritems()}
```

The error is coming from an `assert False` in the parser.  It looks like an issue with the multiple assignment, since wrapping `k,v` in parens works:
```
$ x = {k:float(v) for (k,v) in x.items()}
```

**Comment by wrywerytwreywery** [comment]
Also an issue in list comprehensions:
```
$ x = [(k,float(v)) for k,v in x.items()]
Traceback (most recent call last):
  File "/usr/local/bin/xonsh", line 3, in <module>
    main()
  File "/usr/local/lib/python3.4/site-packages/xonsh/main.py", line 68, in main
    shell.cmdloop()
  File "/usr/local/lib/python3.4/site-packages/xonsh/shell.py", line 181, in cmdloop
    super(Shell, self).cmdloop(intro=intro)
  File "/usr/local/lib/python3.4/cmd.py", line 138, in cmdloop
    stop = self.onecmd(line)
  File "/usr/local/lib/python3.4/cmd.py", line 211, in onecmd
    return self.default(line)
  File "/usr/local/lib/python3.4/site-packages/xonsh/shell.py", line 127, in default
    code = self.push(line)
  File "/usr/local/lib/python3.4/site-packages/xonsh/shell.py", line 152, in push
    locs=self.ctx)
  File "/usr/local/lib/python3.4/site-packages/xonsh/execer.py", line 92, in compile
    tree = self.parse(input, ctx, mode=mode)
  File "/usr/local/lib/python3.4/site-packages/xonsh/execer.py", line 70, in parse
    tree = self._parse_ctx_free(input, mode=mode)
  File "/usr/local/lib/python3.4/site-packages/xonsh/execer.py", line 148, in _parse_ctx_free
    debug_level=self.debug_level)
  File "/usr/local/lib/python3.4/site-packages/xonsh/parser.py", line 258, in parse
    tree = self.parser.parse(input=s, lexer=self.lexer, debug=debug_level)
  File "/usr/local/lib/python3.4/site-packages/ply/yacc.py", line 265, in parse
    return self.parseopt_notrack(input,lexer,debug,tracking,tokenfunc)
  File "/usr/local/lib/python3.4/site-packages/ply/yacc.py", line 971, in parseopt_notrack
    p.callable(pslice)
  File "/usr/local/lib/python3.4/site-packages/xonsh/parser.py", line 2049, in p_comp_for
    assert False
AssertionError
```

**Comment by scopatz** [comment]
OK, I'll try to take a look later tonight.

**Comment by scopatz** [comment]
Fix in #211
